pilot/pkg/model/jwks_resolver.go

```diff
@@ -340,21 +339,27 @@ func (r *JwksResolver) refresher() {
 	for {
 		select {
 		case <-r.refreshTicker.C:
-			r.refresh()
+			refreshInterval := r.refresh()
+			// update refresh interval on change.
+			if r.refreshInterval != refreshInterval {
+				r.refreshTicker.Stop()
+				r.refreshTicker = time.NewTicker(refreshInterval)
```

Comment:
Following my comment in #30407, I think we should add some limit to this, maybe exponential backoff.

cc @incfly and @howardjohn if you have other suggestions.